src/main/java/gregtech/api/worldgen/populator/SurfaceRockPopulator.java

```diff
@@ -84,12 +99,32 @@ public void populateChunk(World world, int chunkX, int chunkZ, Random random, Or
                 int randomZ = chunkZ * 16 + 8 + random.nextInt(16);
                 BlockPos topBlockPos = world.getTopSolidOrLiquidBlock(new BlockPos(randomX, 0, randomZ));
                 Block blockAtPos = world.getBlockState(topBlockPos).getBlock();
-                if ((world.isAirBlock(topBlockPos) || (blockAtPos.isReplaceable(world, topBlockPos) && !world.getBlockState(topBlockPos).getMaterial().isLiquid()))
-                        && world.isSideSolid(topBlockPos.down(), EnumFacing.UP)) {
-                    setStoneBlock(world, topBlockPos, undergroundMaterials);
+
+                //Checks if there is an air block for the surface rock indicator to spawn at
+                if(!world.isAirBlock(topBlockPos)) {
+                    continue;
                 }
+
+                //Checks if the block is a replaceable feature like grass or snow layers. Liquids are replaceable, so
+                // exclude one deep liquid blocks, for looks
+                if(!blockAtPos.isReplaceable(world, topBlockPos) || world.getBlockState(topBlockPos).getMaterial().isLiquid()) {
```

Comment:
Shouldn't this `isReplacable()` check be done above instead of `isAirBlock()`, so that surface rocks will be able to spawn in place of Tall Grass, Snow, etc.?

Reply:
I removed the isAirBlock check entirely, as it was redundant as the Air block is replaceable.